### PR TITLE
rework active+archiving

### LINKED ISSUE #2497: Rework archiving
Archiving is currently both automatic and manual, which sometimes interacts in a surprising fashion.
In a first step, a deterministic, stable behaviour should be created by making archiving fully manual such that no desired operations vanish from sight, neither archived operations become activated unitentionally.

### PR DESCRIPTION
**Pull request by joernu76**
Fix #2497

### CODE REVIEW

**Comment by joernu76** [comment]
This is a first draft to simplify things. A cronjob or similar *could* do an automated archiving on server side, but I dislike this unless we can replace the active flag but a status enum which could give us a further state like "keep".

1) So, can we "simply" replace the "active" bool by a "status" int/enum?
2) I am unsure if/how a change to the status will cause an update of the operation lists of users. Would update_operation need to trigger something here, or is this the job of the function that called update_operation?

**Comment by ReimarBauer** [comment]
@joernu76 have a look at https://github.com/kbr/autocron 
I helped packaging this for conda-forge

**Comment by ReimarBauer** [comment]
Please add also a line to the docs about the archive feature.
https://github.com/Open-MSS/MSS/blob/develop/docs/mscolab.rst?plain=1#L200

**Comment by ReimarBauer** [comment]
> 2\. I am unsure if/how a change to the status will cause an update of the operation lists of users. Would update_operation need to trigger something here, or is this the job of the function that called update_operation?

with reload_operations the client updates its list. 

The server can call update_operation_list. This emits 'operation-list-update'. The client receives this and makes an self.signal_operation_list_updated.emit() The signal is on the client connected to self.reload_operation_list which then triggers reload_operations

**Comment by joernu76** [comment]
> Please add also a line to the docs about the archive feature. https://github.com/Open-MSS/MSS/blob/develop/docs/mscolab.rst?plain=1#L200

done

**Comment by joernu76** [comment]
> > 2. I am unsure if/how a change to the status will cause an update of the operation lists of users. Would update_operation need to trigger something here, or is this the job of the function that called update_operation?
> 
> with reload_operations the client updates its list.
> 
> The server can call update_operation_list. This emits 'operation-list-update'. The client receives this and makes an self.signal_operation_list_updated.emit() The signal is on the client connected to self.reload_operation_list which then triggers reload_operations

I will have a closer look at a couple of interactions between "normal" and "mscolab" in the next weeks, and maybe integrate both into a single class because the split is causing a couple of issues, currently. I hope that this can also be done in smaller steps.

File: mslib/msui/mscolab.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def __init__(self, parent=None, mscolab=None):
 
     def select_archived_operation(self, item):
         logging.debug('select_inactive_operation')
-        if item.access_level == "creator":
+        if item.access_level in ["creator", "admin"]:
```
Comment (ReimarBauer) [inline mslib/msui/mscolab.py:90]: this change must also be done in the UI too, or an admin cannot click the Operation - Maintenance - Archiv in the menu


need_changes_to_activate_button

Comment (ReimarBauer) [reply mslib/msui/mscolab.py:90]: There you need to enable the functions on the given role
https://github.com/Open-MSS/MSS/blob/develop/mslib/msui/mscolab.py#L1967

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
see comment

**Comment by joernu76** [comment]
nice catch

File: mslib/msui/mscolab.py (hunk 1 above)
Comment (joernu76) [reply mslib/msui/mscolab.py:90]: nice catch

**Review by ReimarBauer — APPROVED** [review]
I found a problem for the "other" users  not seeing on their interface that an operation they have open got archieved. As long they are connected they are doing changes.

That maybe worked on in a different PR

archive_switch_does_not_trigger_the_default_operation

The Active Users label sees 2 connected and also the chat. 

image

**Comment by joernu76** [comment]
I would agree. Please open an issue for this. I would assume that a review for the general behaviour due to server side changes (e.g. change of the category of an operation) would be in order.

**Pull request merged**